src/segments/nba.go

```diff
@@ -0,0 +1,438 @@
+		if game.VtAbbreviation == teamTricode || game.HtAbbreviation == teamTricode {
+			return &game, nil
+		}
+	}
+	return nil, errors.New("no scheduled game found for team")
+}
+
+// parses the time and date from the schedule endpoint into a UTC time
+func (nba *NBA) parseTimetoUTC(timeEST string, date string) string {
+	combinedTime := date + " " + timeEST
+	timeUTC, err := time.Parse("01/02/2006 03:04 PM", combinedTime)
+	if err != nil {
+		return ""
+	}
+	return timeUTC.UTC().Format("2006-01-02T15:04:05Z")
+}
+
+// retrieves data from the score endpoint
+func (nba *NBA) retrieveScoreData(teamName string, httpTimeout int) (*NBAData, error) {
+	body, err := nba.env.HTTPRequest(NBAScoreURL, nil, httpTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	var scoreboardResponse *ScoreboardResponse
+	err = json.Unmarshal(body, &scoreboardResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	gameInfo, err := nba.findGameScoreByTeamTricode(scoreboardResponse.Scoreboard.Games, teamName)
+	if err != nil {
+		return nil, err
+	}
+
+	data := NBAData{}
+	data.AwayTeam = gameInfo.AwayTeam.TeamTricode
+	data.HomeTeam = gameInfo.HomeTeam.TeamTricode
+	data.Time = gameInfo.GameStatusText
+	data.GameDate = scoreboardResponse.Scoreboard.GameDate
+	data.StartTimeUTC = gameInfo.GameTimeUTC
+	data.GameStatus = GameStatus(gameInfo.GameStatus)
+	data.HomeScore = gameInfo.HomeTeam.Score
+	data.AwayScore = gameInfo.AwayTeam.Score
+	data.HomeTeamWins = gameInfo.HomeTeam.Wins
+	data.HomeTeamLosses = gameInfo.HomeTeam.Losses
+	data.AwayTeamWins = gameInfo.AwayTeam.Wins
+	data.AwayTeamLosses = gameInfo.AwayTeam.Losses
+
+	return &data, nil
+}
+
+// Retrieves the data from the schedule endpoint
+func (nba *NBA) retrieveScheduleData(teamName string, httpTimeout int) (*NBAData, error) {
+	// How many days into the future should we look for a game.
+	numDaysToSearch := nba.props.GetInt(DaysOffset, 8)
+	nbaSeason := nba.props.GetString(NBASeason, "2023")
+	// Get the current date in America/New_York
+	t := time.Now().In(time.FixedZone("America/New_York", -5*60*60))
+
+	// Check to see if a game is scheduled while the numDaysToSearch is greater than 0
+	for numDaysToSearch > 0 {
+		// convert t into the format that the API expects "MM/YYYY/DD"
+		dateStr := fmt.Sprintf("%02d/%02d/%d", t.Month(), t.Day(), t.Year())
+		urlEndpoint := NBASchedURLPart1 + nbaSeason + "&Date=" + dateStr + NBASchedURLPart2
+		body, err := nba.env.HTTPRequest(urlEndpoint, nil, httpTimeout)
+		if err != nil {
+			return nil, err
+		}
+
+		var scheduleResponse *ScheduleResponse
+		err = json.Unmarshal(body, &scheduleResponse)
+		if err != nil {
+			return nil, err
+		}
+
+		// Check if we can find a game for the team
+		gameInfo, err := nba.findGameSchedulebyTeamTricode(scheduleResponse.ResultSets[1].CompleteGameList, teamName)
+		if err != nil {
+			// We didn't find a game for the team on this day, so we need to check the next day
+			t = t.AddDate(0, 0, 1)
+			numDaysToSearch--
+			continue
+		} else {
```

Comment:
This `else` is redundant. If we hit the first `if`, we will never need this `else` due to the `continue`, so we can simply remove that and "continue" on the same level below.

Reply:
Good catch :)